service/src/test/java/bio/terra/workspace/pact/ProfileApiTest.java

```diff
@@ -0,0 +1,108 @@
+package bio.terra.workspace.pact;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import au.com.dius.pact.consumer.MockServer;
+import au.com.dius.pact.consumer.dsl.PactDslJsonBody;
+import au.com.dius.pact.consumer.dsl.PactDslWithProvider;
+import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
+import au.com.dius.pact.consumer.junit5.PactTestFor;
+import au.com.dius.pact.core.model.RequestResponsePact;
+import au.com.dius.pact.core.model.annotations.Pact;
+import bio.terra.workspace.app.configuration.external.SpendProfileConfiguration;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.spendprofile.SpendProfileId;
+import bio.terra.workspace.service.spendprofile.SpendProfileService;
+import bio.terra.workspace.service.spendprofile.exceptions.SpendUnauthorizedException;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+
+@Tag("pact-test")
+@ExtendWith(PactConsumerTestExt.class)
+public class ProfileApiTest {
+
+  // This will need to be set up with BPM
+  static final String billingProfileId = "4a5afeaa-b3b2-fa51-8e4e-9dbf294b7837";
+
+  @Pact(consumer = "wsm-consumer", provider = "bpm-provider")
+  public RequestResponsePact billingProfileAvailable(PactDslWithProvider builder) {
+    // Using the WSM version of this, instead of the BPM version,
+    // because what we care about is not what BPM sends, but that WSM can handle the value
```

Comment:
Wouldn't we care that WSM can actually handle what BPM sends? Or is that enforced by provider-side tests? In other words, this defines what WSM _can_ handle, and the provider tests enforce that BPM sends something that fits the parameters defined by WSM?

Reply:
Basically, yes.
This defines the pact, which is then run against the provider. 
So we're saying basically "we can handle 'GCP' and 'Azure'". I defined it like this to ensure the string values match up. 

Note that I split this into two tests, one for GCP and one for Azure, after we determined last week that the old test setup is demanding mutually contradictory values.